jib-core/src/main/java/com/google/cloud/tools/jib/configuration/LayerConfiguration.java

```diff
@@ -103,12 +99,17 @@ public Builder addEntry(
      *     sourceFile}
      * @param permissions the file permissions on the container. If null, then default permissions
      *     are used (644 for files, 755 for directories)
+     * @param lastModified the provider of the file modification timestamp
      * @return this
      * @see Builder#addEntry(Path, AbsoluteUnixPath)
      */
     public Builder addEntry(
-        Path sourceFile, AbsoluteUnixPath pathInContainer, @Nullable FilePermissions permissions) {
-      return addEntry(sourceFile, pathInContainer, permissions, 1000);
+        Path sourceFile,
+        AbsoluteUnixPath pathInContainer,
+        @Nullable FilePermissions permissions,
+        FileTimestampProvider lastModified) {
```

Comment:
As @TadCordle pointed out, considering that `addEntry()` is to add a single file to the layer, it can be confusing from the user perspective that `lastModified` is a function from `Path` to `Instant`. We should clearly document the function is to take the source path. However, it still seems unintuitive given that you are adding a single file and you can always determine the timestamp at the time of using this API. In the end, the function will always get the `sourceFile` parameter being passed here; the function is not for determining timestamps for many different files. Such function will make sense at a higher level when the user is passing multiple files. @briandealwis WDYT?

Reply:
I kinda agree, having a provider overcomplicates things a bit in this case since the timestamp can just be determined when the files are added to the `LayerConfiguration`. I could see it being more useful for something like `addEntryRecursive` where you can't set the timestamp of each individual file; maybe an overload or two for `addEntryRecursive` that take a timestamp/file permission provider would be more useful, and the non-recursive `addEntry` should just take an `Instant`.

Reply:
I agree: a provider seems a burden.  But I do like @TadCordle's use of `Instant` over a long, though maybe we should take both to avoid unnecessary API breakage.  Maybe deprecate the variant with a long?

Reply:
It would only change `LayerEntry`, which isn't intended to be an API class. So I don't think we need to deprecate anything for now.

Reply:
We have an `Builder.addEntry()` that takes a long that [you're proposing to remove a bit above](https://github.com/GoogleContainerTools/jib/pull/1613/files#diff-fcb2b167686b2e075cb5149f1862e28eL87)

Reply:
I wonder if we wouldn't be better extracting `addEntryRecursive` into a separate walker that calls `addEntry`.